### PR TITLE
Bug 2000554: oc inspect: Also gather namespaces in which webhook services are running

### PR DESCRIPTION
**Pull request by ardaguclu**
Currently, `oc inspect` obtains related objects by extracting them from `.status.relatedObjects` field in resources. However, `Validating|MutatingWebhookConfigurations` do not contain this field, instead they are storing services in `webhook.clientConfig.service` field. If services are in another namespace, `oc inspect` does not inspect those namespaces.

This PR adds ability to gather namespaces in which webhook services are running. That will shed more light on troubleshooting.

### CODE REVIEW

File: pkg/cli/admin/inspect/resource.go [hunk 1]
```diff
@@ -184,3 +188,38 @@ func obtainRelatedObjects(obj *unstructured.Unstructured) ([]*configv1.ObjectRef
...
+	var relatedObjs []*configv1.ObjectReference
+	for _, w := range webhooks {
+		w, ok := w.(map[string]interface{})
+		if !ok {
+			continue
+		}
+		ns, found, err := unstructured.NestedString(w, "clientConfig", "service", "namespace")
+		if !found || err != nil {
+			klog.V(1).Infof("%q does not contain .webhook.clientConfig.service.namespace", unstructuredToString(obj))
+			return nil
```
[the remark this answers is not in the document]
Comment (ardaguclu) [reply pkg/cli/admin/inspect/resource.go:212]: 1 minute ago, I was preparing a new push to change this :). Thanks

**Review by ingvagabund — APPROVED** [review]

**Comment by ingvagabund** [comment]
@kasturinarra would you please perform per-merge testing?

File: pkg/cli/admin/inspect/resource.go [hunk 2]
```diff
@@ -166,6 +166,10 @@ func obtainRelatedObjects(obj *unstructured.Unstructured) ([]*configv1.ObjectRef
 	// obtain related namespace info for the current resource
 	klog.V(1).Infof("Gathering related object reference information for %q...\n", unstructuredToString(obj))
 
+	if obj.GetKind() == "ValidatingWebhookConfiguration" || obj.GetKind() == "MutatingWebhookConfiguration" {
```
Comment (soltysh) [inline pkg/cli/admin/inspect/resource.go:169]: Nit: let's be explicit and move this after this block:https://github.com/openshift/oc/blob/0703a5baabaf66ca39604cfb655fd854d991b294/pkg/cli/admin/inspect/resource.go#L124-L127

This way we'll have clean `obtainReleatedObjects` and `obtainWebhookService` as two steps in `gatherRelatedObjects`. In the future, if someone decides to add `.status.relatedObject` to webhooks we'll still have this mechanism working.

Comment (soltysh) [reply pkg/cli/admin/inspect/resource.go:169]: `obtainWebhookService` is a rename of your `obtainWebhookRelatedObjects`

**Review by soltysh — CHANGES_REQUESTED** [review]
/lgtm cancel

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by ardaguclu** [comment]
/hold

to see @kasturinarra's manual testing.

**Pull request merged**